### DISCUSSION TITLE
Select component empty state handling

### DISCUSSION THREAD

**Discussion by heyethankim** [patternfly/patternfly#5575] [Feature requests]
## Requesting new features, enhancements, or design changes to PatternFly

When the options in the [select component](https://www.patternfly.org/v4/components/select/react-next/) are already available, it is very straightforward.

But what if the options are generated by user? For example, in the ACM ApplicationSet creation wizard, “Argo Server” is empty at first. If we know it is empty at first, we should provide a different experience.

<img width="1132" alt="image" src="https://user-images.githubusercontent.com/84546841/236473692-fbec44bf-8be9-4c00-98d7-a54c00a91a85.png">

**Comment by mcarrano** [comment]
Converting for more discussion...

**Comment by mmenestr** [comment]
An idea -- we could show the menu toggle, but when clicked on, the user would see an extra small empty state in the menu with a link to "Add Argo server", which would link out to a new tab for modal wizards, or open a modal for in page wizard or forms.

We could document this either in the empty state guidelines or in a new "Patterns" section

**Comment by heyethankim** [comment]
Margot, you shared a few options to tackle the issue, and the ultimate question I got was what should trigger the action. The fact I don't have a thing to continue OR the component you should have when you have a thing.

**Reply by mmenestr** [reply]
I'm not sure I understand. Isn't the link what triggers the action? What action are you referring to?
